pkg/runner/logger.go

```diff
@@ -46,39 +66,46 @@ func WithJobLogger(ctx context.Context, jobName string, config *Config, masks *[
 	if config.JSONLogger {
 		formatter = &jobLogJSONFormatter{
 			formatter: &logrus.JSONFormatter{},
-			masker:    valueMasker(config.InsecureSecrets, config.Secrets, masks),
+			masker:    valueMasker(config.InsecureSecrets, config.Secrets),
 		}
 	} else {
 		formatter = &jobLogFormatter{
 			color:  colors[nextColor%len(colors)],
-			masker: valueMasker(config.InsecureSecrets, config.Secrets, masks),
+			masker: valueMasker(config.InsecureSecrets, config.Secrets),
 		}
 	}
 
-	nextColor++
+	ctx = WithMasks(ctx, masks)
```

Comment:
> coloring does work

You disabled coloring all jobs are blue for me, by never doing nextcolor++. You are correct that composite actions have the same color in this PR.
```suggestion
	nextColor++
	ctx = WithMasks(ctx, masks)
```